GenderPayGap.WebUI/Models/ManageOrganisations/ManageOrganisationViewModel.cs

```diff
@@ -101,36 +106,125 @@ public string GetRequiredToReportOrNotText()
             {
-                DateTime snapshotDate = organisation.SectorType.GetAccountingStartDate(ReportingYear);
-                DateTime deadline = ReportingYearsHelper.GetDeadlineForAccountingDate(snapshotDate);
+                return ReportStatus.Due;
+            }
 
-                return "by " + deadline.ToString("d MMM yyyy");
+            if (OrganisationIsRequiredToSubmit() && reportIsNotSubmitted && DeadlineHasPassed())
+            {
+                return ReportStatus.Overdue;
             }
 
-            return null;
+            if (!reportIsNotSubmitted
+                && (returnForYear.IsRequired() && returnForYear.IsSubmittedOnTime()
+                    || !returnForYear.IsRequired() && returnForYear.IsSubmitted()))
+            {
+                return ReportStatus.Submitted;
+            }
+
+            if (!reportIsNotSubmitted && returnForYear.IsRequired() && returnForYear.IsLateSubmission)
+            {
+                return ReportStatus.SubmittedLate;
+            }
+            
+            return ReportStatus.NotRequired;
+
         }
 
         public string GetReportStatusText()
         {
+            switch (GetReportStatus())
+            {
+                case ReportStatus.Due:
+                    return "Report due by " + GetReportingDeadline().ToString("d MMMM yyyy");
+                case ReportStatus.Overdue:
+                    return "Report overdue";
+                case ReportStatus.Submitted:
+                    return "Report submitted";
+                case ReportStatus.SubmittedLate:
+                    return "Report submitted late";
+                default:
+                    return "Report not required";
+            }
+        }
+        
+        public string GetReportStatusColour()
+        {
+            switch (GetReportStatus())
+            {
+                case ReportStatus.Due:
+                    return "govuk-tag--blue";
+                case ReportStatus.Overdue:
+                    return "govuk-tag--red";
+                case ReportStatus.Submitted:
+                case ReportStatus.SubmittedLate:
+                    return "govuk-tag--green";
+                default:
+                    return "govuk-tag--grey";
+            }
+        }
+
+        public string GetReportStatusDescription()
+        {
+            ReportStatus status = GetReportStatus();
             Return returnForYear = organisation.GetReturn(ReportingYear);
 
-            if (returnForYear == null)
+            switch (status)
             {
-                return "Your employer has not reported";
+                case ReportStatus.Overdue:
+                    return "This report was due on " + GetReportingDeadline().ToString("d MMMM yyyy");
+                case ReportStatus.Submitted:
+                case ReportStatus.SubmittedLate:
+                    return "Reported on " + returnForYear.Modified.ToString("d MMMM yyyy");

```

Comment:
Is there a reason why you didn't keep `new GDSDateFormatter(returnForYear.Modified).FullStartDate`?

Reply:
I wasn't sure about creating a new object every time we needed to format a date

Reply:
Creating new objects is not that expensive if there aren't a lot so I'd rather use that. If the date format needs to be changed across the service we have to update it in a single place.